File: chapter-05/01_gcd_with_euclid_algorithm.c
# include <stdio.h>

int eucild_gcd(int a, int b) { 
/*
    Greatest Common Divisor(gcd) or গসাগু calcultion useing Euclid algorithm
    Euclid algorithm:-
        a ও b-এর গসাগু = (a - b) ও b-এর গসাগু
*/
    int temp;
    if (b > a) {
        temp = b;
        b = a;
        a = temp;
    }
    while (a != b) {
        if (b == 0) {
            break;
        }
        if (a > b) {
            a = a - b;
        }
        else {
            b = b - a;
        }
    }
    return a; // it return gcd of a and b
}

int main() {
    int a, b;

    printf("Enter the first number\t: ");
    scanf("%d", &a);
    printf("Enter the second number\t: ");
    scanf("%d", &b);

    
    printf("GCD is %d\n", eucild_gcd(a, b));


    return 0;
}

File: gonit-korbo-joi_tamim-shahriar-subeen/chapter-05/01_gcd_with_euclid_algorithm.py
a = input("Enter the first number: ")
a = int(a)
b = input("Enter the second number: ")
b = int(b)

# main algorithm
# a ও b-এর গসাগু = (a - b) ও b-এর গসাগু 

if b > a:
    #b যদি a-এর চেয়ে বড় হয়, তাহলো তাদেরকে অদলবদল করে দিচ্ছি
    a, b = b, a

while a != b:
    if b == 0: #b-এর মান 0 হলে, গসাগু হবে a, তাই আমরা লুপ থেকে বের হয়ে যাব
        break
    if a > b:
        a = a - b
    else:
        b = b - a

print("GCD is", a)